TestAssets/TestProjects/MSBuildTestEchoArgsApp/MSBuildTestApp.csproj

```diff
@@ -0,0 +1,16 @@
+<Project Sdk="Microsoft.NET.Sdk" ToolsVersion="15.0" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
```

Comment:
Can't you just modify the existing MSBuildTestApp slightly instead of creating a new one.

You can do something like, if args.Length > 0, do this behavior. Otherwise keep the current one.